pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -93,6 +166,13 @@ func (ctrl *Controller) syncNodeConfigHandler(key string) error {
 		return fmt.Errorf("could not get the TLSSecurityProfile from %v: %v", ctrlcommon.APIServerInstanceName, err)
 	}
 
+	for _, pool := range mcpPools {
+		// First, create the auto-sizing MachineConfig for this pool if it doesn't exist
+		if err := ctrl.configureAutoSizingMachineConfig(pool); err != nil {
+			return err
```

Comment:
I know that this function `syncNodeConfigHandler` would always get hit as we have a `nodes.config` object in the cluster always
IMHO, the creation of this MC is not related to the `nodes.config.openshift.io` object in anyway and hence this is not the correct place to have this operation.
For ex, encountering an issue in creating an ignition file or a machineconfig should not interrupt the functionality of `nodes.config` api i.e. updating the `workerLatencyProfiles` or the `minKubeletVersion` etc.
The only place we should deal with is the `kubeletConfigController`. We can have an extra sync/work item which can be hit always without depending on the `KubeletConfig` object's presence.
Could you explore in that direction?